lib/poller.js

```diff
@@ -64,16 +66,40 @@ class Poller extends EventEmitter {
         acc[preloadKey] = latestCache[cacheKey]
         // Serialize values for splits and segments
         if (['splitsData', 'segmentsData'].includes(preloadKey)) {
-          const data = acc[preloadKey]
-          Object.entries(data).forEach(([key, value]) => { data[key] = typeof value === 'string' ? value : JSON.stringify(value) })
-          acc[preloadKey] = data
+          const processedEntries = Object.entries(acc[preloadKey])
```

Comment:
```suggestion
          acc[preloadKey] = Object.entries(acc[preloadKey])
```
Since we only use this const once?